src/utils.js

```diff
@@ -1605,6 +1611,25 @@ _.info = {
     }
 };
 
+// naive way to extract domain name (example.com) from full hostname (my.sub.example.com)
+var SIMPLE_DOMAIN_MATCH_REGEX = /[a-z0-9][a-z0-9-]*\.[a-z]+$/i;
+// this next one attempts to account for some ccSLDs, e.g. extracting oxford.ac.uk from www.oxford.ac.uk
+// it's a pretty blunt heuristic and you can't do this reliably without a list like at https://publicsuffix.org/
+// but it maintains backwards compatibility with existing Mixpanel integrations
+var DOMAIN_MATCH_REGEX = /[a-z0-9][a-z0-9-]+\.[a-z.]{2,6}$/i;
+var has_long_tld = function(hostname) {
+    var parts = hostname.split('.');
+    return parts[parts.length - 1].length > 4;
+};
+var extract_domain = function(hostname) {
+    var domain_regex = DOMAIN_MATCH_REGEX;
+    if (hostname.endsWith('.com') || hostname.endsWith('.org') || has_long_tld(hostname)) {
+        domain_regex = SIMPLE_DOMAIN_MATCH_REGEX;
+    }
+    var matches = hostname.match(domain_regex);
+    return matches ? matches[0] : '';
+};
```

Comment:
I'd love to see a short comment at the top of this code block summarizing the current set of heuristics used. You can parse the intent out of the code, but a succinct description in english would go a long way towards speeding up and ensuring correct comprehension.

Reply:
https://github.com/mixpanel/mixpanel-js/pull/251/commits/f4f78827d2f1260b5afd71739111b6acab8edc3f